src/Microsoft.ApplicationInsights.AspNet/Extensibility/Implementation/Tracing/AspNetEventSource.cs

```diff
@@ -0,0 +1,121 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AspNetEventSource.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+namespace Microsoft.ApplicationInsights.AspNet.Extensibility.Implementation.Tracing
+{
+    using System;
+    using System.Diagnostics.Tracing;
+
+    /// <summary>
+    /// Event source for Application Insights ASP.NET 5 SDK.
+    /// </summary>
+    [EventSource(Name = "Microsoft-ApplicationInsights-AspNet")]
+    internal sealed class AspNetEventSource : EventSource
```

Comment:
There are two very different paradigms between ILogger and EventSource.  I can make it work the way you describe but it would not be following the ILogger model and there are several reasons why I would not recommend it we can discuss.